File: dataset/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Task132.java

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.*;
import java.io.*;

class SecurityEvent {
    private String eventType;
    private String severity;
    private String source;
    private String description;
    private LocalDateTime timestamp;
    
    public SecurityEvent(String eventType, String severity, String source, String description) {
        this.eventType = eventType;
        this.severity = severity;
        this.source = source;
        this.description = description;
        this.timestamp = LocalDateTime.now();
    }
    
    public String getEventType() { return eventType; }
    public String getSeverity() { return severity; }
    public String getSource() { return source; }
    public String getDescription() { return description; }
    public LocalDateTime getTimestamp() { return timestamp; }
    
    @Override
    public String toString() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        return String.format("[%s] [%s] [%s] %s - %s",
            timestamp.format(formatter), severity, eventType, source, description);
    }
}

class SecurityMonitor {
    private List<SecurityEvent> eventLog;
    private Map<String, Integer> eventCounts;
    private Map<String, Integer> severityCounts;
    
    public SecurityMonitor() {
        this.eventLog = new ArrayList<>();
        this.eventCounts = new HashMap<>();
        this.severityCounts = new HashMap<>();
    }
    
    public void logEvent(String eventType, String severity, String source, String description) {
        SecurityEvent event = new SecurityEvent(eventType, severity, source, description);
        eventLog.add(event);
        
        eventCounts.put(eventType, eventCounts.getOrDefault(eventType, 0) + 1);
        severityCounts.put(severity, severityCounts.getOrDefault(severity, 0) + 1);
        
        System.out.println("Logged: " + event);
    }
    
    public List<SecurityEvent> getEventsBySeverity(String severity) {
        List<SecurityEvent> result = new ArrayList<>();
        for (SecurityEvent event : eventLog) {
            if (event.getSeverity().equals(severity)) {
                result.add(event);
            }
        }
        return result;
    }
    
    public List<SecurityEvent> getEventsByType(String eventType) {
        List<SecurityEvent> result = new ArrayList<>();
        for (SecurityEvent event : eventLog) {
            if (event.getEventType().equals(eventType)) {
                result.add(event);
            }
        }
        return result;
    }
    
    public Map<String, Integer> getEventStatistics() {
        return new HashMap<>(eventCounts);
    }
    
    public Map<String, Integer> getSeverityStatistics() {
        return new HashMap<>(severityCounts);
    }
    
    public List<SecurityEvent> getAllEvents() {
        return new ArrayList<>(eventLog);
    }
    
    public int getTotalEvents() {
        return eventLog.size();
    }
    
    public void generateReport() {
        System.out.println("\\n=== Security Event Report ===");
        System.out.println("Total Events: " + getTotalEvents());
        System.out.println("\\nEvents by Type:");
        for (Map.Entry<String, Integer> entry : eventCounts.entrySet()) {
            System.out.println("  " + entry.getKey() + ": " + entry.getValue());
        }
        System.out.println("\\nEvents by Severity:");
        for (Map.Entry<String, Integer> entry : severityCounts.entrySet()) {
            System.out.println("  " + entry.getKey() + ": " + entry.getValue());
        }
        System.out.println("============================\\n");
    }
}

public class Task132 {
    public static void main(String[] args) {
        SecurityMonitor monitor = new SecurityMonitor();
        
        // Test Case 1: Log authentication failure
        System.out.println("Test Case 1: Authentication Failure");
        monitor.logEvent("AUTH_FAILURE", "HIGH", "192.168.1.100", "Failed login attempt for user 'admin'");
        
        // Test Case 2: Log successful login
        System.out.println("\\nTest Case 2: Successful Login");
        monitor.logEvent("AUTH_SUCCESS", "INFO", "192.168.1.101", "User 'john.doe' logged in successfully");
        
        // Test Case 3: Log unauthorized access attempt
        System.out.println("\\nTest Case 3: Unauthorized Access");
        monitor.logEvent("UNAUTHORIZED_ACCESS", "CRITICAL", "192.168.1.105", "Attempt to access restricted resource");
        
        // Test Case 4: Log multiple events and generate report
        System.out.println("\\nTest Case 4: Multiple Events");
        monitor.logEvent("AUTH_FAILURE", "HIGH", "192.168.1.102", "Brute force attack detected");
        monitor.logEvent("DATA_ACCESS", "MEDIUM", "192.168.1.103", "Sensitive data accessed by user 'jane.smith'");
        monitor.logEvent("AUTH_FAILURE", "HIGH", "192.168.1.100", "Another failed login for 'admin'");
        
        monitor.generateReport();
        
        // Test Case 5: Query events by severity
        System.out.println("Test Case 5: Query High Severity Events");
        List<SecurityEvent> highSeverityEvents = monitor.getEventsBySeverity("HIGH");
        System.out.println("High Severity Events: " + highSeverityEvents.size());
        for (SecurityEvent event : highSeverityEvents) {
            System.out.println("  " + event);
        }
        
        System.out.println("\\nQuery Events by Type (AUTH_FAILURE):");
        List<SecurityEvent> authFailures = monitor.getEventsByType("AUTH_FAILURE");
        System.out.println("Authentication Failures: " + authFailures.size());
    }
}

File: dataset/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Task132.py

from datetime import datetime
from collections import defaultdict
from typing import List, Dict

class SecurityEvent:
    def __init__(self, event_type, severity, source, description):
        self.event_type = event_type
        self.severity = severity
        self.source = source
        self.description = description
        self.timestamp = datetime.now()
    
    def __str__(self):
        timestamp_str = self.timestamp.strftime("%Y-%m-%d %H:%M:%S")
        return f"[{timestamp_str}] [{self.severity}] [{self.event_type}] {self.source} - {self.description}"

class SecurityMonitor:
    def __init__(self):
        self.event_log = []
        self.event_counts = defaultdict(int)
        self.severity_counts = defaultdict(int)
    
    def log_event(self, event_type, severity, source, description):
        event = SecurityEvent(event_type, severity, source, description)
        self.event_log.append(event)
        
        self.event_counts[event_type] += 1
        self.severity_counts[severity] += 1
        
        print(f"Logged: {event}")
        return event
    
    def get_events_by_severity(self, severity):
        return [event for event in self.event_log if event.severity == severity]
    
    def get_events_by_type(self, event_type):
        return [event for event in self.event_log if event.event_type == event_type]
    
    def get_event_statistics(self):
        return dict(self.event_counts)
    
    def get_severity_statistics(self):
        return dict(self.severity_counts)
    
    def get_all_events(self):
        return self.event_log.copy()
    
    def get_total_events(self):
        return len(self.event_log)
    
    def generate_report(self):
        print("\\n=== Security Event Report ===")
        print(f"Total Events: {self.get_total_events()}")
        print("\\nEvents by Type:")
        for event_type, count in self.event_counts.items():
            print(f"  {event_type}: {count}")
        print("\\nEvents by Severity:")
        for severity, count in self.severity_counts.items():
            print(f"  {severity}: {count}")
        print("============================\\n")

def main():
    monitor = SecurityMonitor()
    
    # Test Case 1: Log authentication failure
    print("Test Case 1: Authentication Failure")
    monitor.log_event("AUTH_FAILURE", "HIGH", "192.168.1.100", "Failed login attempt for user 'admin'")
    
    # Test Case 2: Log successful login
    print("\\nTest Case 2: Successful Login")
    monitor.log_event("AUTH_SUCCESS", "INFO", "192.168.1.101", "User 'john.doe' logged in successfully")
    
    # Test Case 3: Log unauthorized access attempt
    print("\\nTest Case 3: Unauthorized Access")
    monitor.log_event("UNAUTHORIZED_ACCESS", "CRITICAL", "192.168.1.105", "Attempt to access restricted resource")
    
    # Test Case 4: Log multiple events and generate report
    print("\\nTest Case 4: Multiple Events")
    monitor.log_event("AUTH_FAILURE", "HIGH", "192.168.1.102", "Brute force attack detected")
    monitor.log_event("DATA_ACCESS", "MEDIUM", "192.168.1.103", "Sensitive data accessed by user 'jane.smith'")
    monitor.log_event("AUTH_FAILURE", "HIGH", "192.168.1.100", "Another failed login for 'admin'")
    
    monitor.generate_report()
    
    # Test Case 5: Query events by severity
    print("Test Case 5: Query High Severity Events")
    high_severity_events = monitor.get_events_by_severity("HIGH")
    print(f"High Severity Events: {len(high_severity_events)}")
    for event in high_severity_events:
        print(f"  {event}")
    
    print("\\nQuery Events by Type (AUTH_FAILURE):")
    auth_failures = monitor.get_events_by_type("AUTH_FAILURE")
    print(f"Authentication Failures: {len(auth_failures)}")

if __name__ == "__main__":
    main()
